src/test/java/org/generationcp/middleware/api/location/LocationServiceImplIntegrationTest.java

```diff
@@ -52,6 +54,38 @@ public void testGetFilteredLocations() {
 
 	}
 
+	@Test
+	public void testcountLocations() throws MiddlewareQueryException {
+		final LocationSearchRequest locationSearchRequest = new LocationSearchRequest();
+		locationSearchRequest.setLocationTypeName("COUNTRY");
+		final long countLocation = this.locationService.countLocations(locationSearchRequest);
+		assertThat("Expected country location size > zero", 0 < countLocation);
+	}
+
+	@Test
+	public void testcountLocationsByFilterNotRecoveredLocation() throws MiddlewareQueryException {
+		final LocationSearchRequest locationSearchRequest = new LocationSearchRequest();
+		locationSearchRequest.setLocationTypeName("DUMMYLOCTYOE");
+		final long countLocation = this.locationService.countLocations(locationSearchRequest);
+		assertThat("Expected country location size equals to zero", 0 == countLocation);
+	}
+
+	@Test
+	public void testgetLocationsByFilter() throws MiddlewareQueryException {
+		final LocationSearchRequest locationSearchRequest = new LocationSearchRequest();
+		locationSearchRequest.setLocationTypeName("COUNTRY");
+		final List<org.generationcp.middleware.api.location.Location> locationList = this.locationService.getLocations(locationSearchRequest, new PageRequest(0, 10));
+		assertThat("Expected list of location size > zero", !locationList.isEmpty());
+	}
+
+	@Test
+	public void testgetLocationsByFilterNotRecoveredLocation() throws MiddlewareQueryException {
```

Comment:
Remove throwing of `MiddlewareQueryException` in all test methods (no need to throw runtime exceptions)